Sources/SwiftDocC/Infrastructure/Workspace/FeatureFlags+Info.swift

```diff
@@ -0,0 +1,86 @@
+/*
+ This source file is part of the Swift.org open source project
+
+ Copyright (c) 2024 Apple Inc. and the Swift project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+ See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+
+import Foundation
+
+extension DocumentationBundle.Info {
+    /// A collection of feature flags that can be enabled from a bundle's Info.plist.
+    ///
+    /// This is a subset of flags from ``FeatureFlags`` that can influence how a documentation
+    /// bundle is written, and so can be considered a property of the documentation itself, rather
+    /// than as an experimental behavior that can be enabled for one-off builds.
+    ///
+    /// ```xml
+    /// <key>CDExperimentalFeatureFlags</key>
+    /// <dict>
+    ///     <key>ExperimentalOverloadedSymbolPresentation</key>
+    ///     <true/>
+    /// </dict>
+    /// ```
+    public struct BundleFeatureFlags: Codable, Equatable {
+        /// Whether or not experimental support for combining overloaded symbol pages is enabled.
+        ///
+        /// This feature flag corresponds to ``FeatureFlags/isExperimentalOverloadedSymbolPresentationEnabled``.
+        public var experimentalOverloadedSymbolPresentation: Bool?
```

Comment:
Is there a way we can automatically make this synchronized with every flag in `FeatureFlags`?

This would involve reworking `FeatureFlags` so that it exposes the list of allowable keys, but it would be neat to make it possible to add a feature flag just in `FeatureFlags` and automatically get the parsing from Info.plist functionality.

Reply:
I could imagine dropping `BundleFeatureFlags` in favor of making `FeatureFlags` itself be Codable, or generating a setup using key paths from the global `FeatureFlags` (so that the global struct wouldn't need to be changed to use optional types). I feel like i appreciate the deliberate design of making only some feature flags be part of a bundle, though; not every feature flag really warrants updating a bundle's metadata to turn it on, IMO.

Reply:
In either scenario you wouldn't have to update the bundle's metadata if you don't want to, but having the ability to specify any feature flag from either the CLI or the Info.plist seems valuable to me. Especially in the case where clients don't control the DocC invocation (e.g. Swift Package Index) it would be a nice way to let clients customize DocC's behavior.

Reply:
The SPI use case is compelling enough that i'll look into it.

Reply:
It looks like providing _fully automatic_ availability might be a bit more complicated than i'm willing to write into this PR, especially since it's been getting a lot of back-and-forth already. I'd like to work on that as a future enhancement rather than delay this PR even more.

Reply:
If it sounds like we may change this API in the near future that's a great reason to not make it public yet. Otherwise we'll need to keep the old deprecated version around for a while as well.

Reply:
I took the type internal and removed it from public API in [e837c32](https://github.com/apple/swift-docc/pull/891/commits/e837c32ffedb0171a544e4fddfcc89f424b8a9f4). Is this sufficient to allow iteration? I'd like to land this soon since it's been in flight for a while.

Reply:
Yes, that should be sufficient. Note that internal is the default access level, so it doesn't _need_ to be explicitly specified.